src/modules/components/dao/daoAvatar/daoAvatar.tsx

```diff
@@ -0,0 +1,57 @@
+import classNames from 'classnames';
+import type React from 'react';
+import { Avatar, type IAvatarProps } from '../../../../core';
+
+export interface IDaoAvatarProps extends Omit<IAvatarProps, 'fallback'> {
+    /**
+     * Name of the DAO
+     */
+    daoName: string;
+}
+
+export const DaoAvatar: React.FC<IDaoAvatarProps> = (props) => {
+    const { daoName, size = 'lg', className, ...otherProps } = props;
+    const daoInitials = getDaoInitials(daoName).toUpperCase();
+
+    return (
+        <Avatar
+            size={size}
+            fallback={
+                <span
+                    className={classNames(
+                        'flex size-full shrink-0 items-center justify-center rounded-full bg-primary-400',
+                        'text-xs leading-tight text-neutral-0 md:text-sm lg:text-lg',
+                        className,
+                    )}
+                >
+                    {daoInitials}
+                </span>
+            }
+            {...otherProps}
+        />
+    );
+};
+
+/**
+ * Get the initials for a DAO
+ * @param daoName name of the DAO
+ * @returns the DAO initials
+ */
+function getDaoInitials(daoName: IDaoAvatarProps['daoName']): string {
+    if (!daoName) {
+        return '';
+    }
```

Comment:
The `daoName` can't be undefined and the empty string case is already handle below, right?
```suggestion
```

Reply:
`daoName` shouldn't be undefined, but it certainly can be. The DAO name isn't on chain but pinned wherever, so the structure cannot be 100% enforced. Sure the consumer can pass a default empty string, but this ensures the whole application doesn't crash since this string is being processed by the component.

Reply:
Probably the most complete solution, yes.